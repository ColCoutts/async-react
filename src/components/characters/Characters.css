.Characters {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  width: 100vw;
  height: 100vh;
  
  li {
    width: 20vw;
    margin: 2em;
    padding: 2rem;

    background-color: #f0f0f0;
    text-align: center;
    list-style-type: none;
  }

  section {
    margin: auto;
    animation: insert .5s linear;
  }

  img {
    width: 10rem;
    margin: auto;
  }

  @keyframes insert {
    from {
      height: 0;
    }

    to {
      height: 100%
    }
  }
}
