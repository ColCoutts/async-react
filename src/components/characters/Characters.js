import React from 'react';
import PropTypes from 'prop-types';

function Characters({ characters }) {
  const characterList = characters.map((character, i) => {
    
  })
}
