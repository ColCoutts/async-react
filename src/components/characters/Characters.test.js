import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters component', () => {
  it('renders a snapshot', () => {
    const characters = [
      {
        name: 'Colin',
        gender: 'Alient',
        age: '-less',
        films: [
          'this film'
        ]
      },
      {
        name: 'Chad',
        gender: 'Alient',
        age: '-less',
        films: [
          'this film'
        ]
      },
      {
        name: 'Chris',
        gender: 'Alient',
        age: '-less',
        films: [
          'this film'
        ]
      }
    ];

    const wrapper = shallow(<Characters characters={characters} />)
    expect(wrapper).toMatchSnapshot();
  });
});
