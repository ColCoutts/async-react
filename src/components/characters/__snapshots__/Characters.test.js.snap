// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders a snapshot 1`] = `
<ul>
  <li
    key="0"
  >
    <Character
      character={
        Object {
          "age": "-less",
          "films": Array [
            "this film",
          ],
          "gender": "Alient",
          "name": "Colin",
        }
      }
    />
  </li>
  <li
    key="1"
  >
    <Character
      character={
        Object {
          "age": "-less",
          "films": Array [
            "this film",
          ],
          "gender": "Alient",
          "name": "Chad",
        }
      }
    />
  </li>
  <li
    key="2"
  >
    <Character
      character={
        Object {
          "age": "-less",
          "films": Array [
            "this film",
          ],
          "gender": "Alient",
          "name": "Chris",
        }
      }
    />
  </li>
</ul>
`;
